test/unit/ajax.js

```diff
@@ -837,6 +837,41 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - do not execute scripts from unsuccessful responses (gh-4250)", 2, function( assert ) {
```

Comment:
One more thing that I forgot - we could use a test that sets a custom `text script` converter and makes sure it _does not_ get invoked for 404 scripts.